File: 백준/Silver/6603. 로또/로또.py
def make(level, now):
    if level == 6:
        print(*ans)
        return

    for i in range(now, s):  
        ans.append(numbers[i])
        make(level+1, i+1)  
        ans.pop() 

while True:
    numbers = list(map(int, input().split()))
    s = numbers[0]
    if s == 0:
        break
    numbers = numbers[1:]
    ans = []
    make(0, 0)
    print()

File: 백준/Silver/6603. 로또/로또.cc
#include <iostream>
using namespace std;


int k;
int s[13] = { 0, };
int ans[6] = { 0, };

void lotto(int start, int level) {

	if (level == 6) {
		for (int i = 0;i < 6;i++) {
			cout << ans[i] << " ";
		}
		cout << '\n';
		return;
	}


	for (int i = start;i < k;i++) {
		ans[level] = s[i];
		lotto(i+1, level + 1);
	}

}



int main() {


	while (1) {
		cin >> k;
		if (k==0) {
			break;
		}
		for (int i = 0;i < k;i++) {
			cin >> s[i];
		}

		lotto(0, 0);

		cout << '\n';

	}

	return 0;
}